local/gugcat/renderer.php

```diff
@@ -203,8 +203,20 @@ public function display_aggregation_tool($rows, $activities) {
 
         $htmlcolumns = null;
         $htmlrows = null;
+        $weight_arr = array();
+        $index = 0;
+
+        foreach($rows as $r){
+            foreach($r->grades as $grade){
+                array_push($weight_arr, $grade->weight);
+            }
+        }
+    
+        $weights = array_unique($weight_arr);
+
```

Comment:
I'm not sure why you will need this. I think you can directly access the gradebook grades from $act->gradeitem->aggregationcoef

Reply:
You can check the grade_capture->set_provisional_weights() on how we get the weights from each activities.